pandas/io/sql.py

```diff
@@ -692,37 +692,26 @@ def insert_data(self):
         column_names = list(map(str, temp.columns))
         ncols = len(column_names)
         data_list = [None] * ncols
-        blocks = temp._mgr.blocks
-
-        for b in blocks:
-            if b.is_datetime:
-                # return datetime.datetime objects
-                if b.is_datetimetz:
-                    # GH 9086: Ensure we return datetimes with timezone info
-                    # Need to return 2-D data; DatetimeIndex is 1D
-                    d = b.values.to_pydatetime()
-                    d = np.atleast_2d(d)
-                else:
-                    # convert to microsecond resolution for datetime.datetime
-                    d = b.values.astype("M8[us]").astype(object)
-            elif b.is_timedelta:
-                # numpy converts this to an object array of integers,
-                #  whereas b.astype(object).values would convert to
-                #  object array of Timedeltas
-                d = b.values.astype(object)
+
+        for i in range(len(temp.columns)):
+            ser = temp.iloc[:, i]
```

Comment:
you should just use .items() here no? which yields the name and column

Reply:
we're going to need `i` below

Reply:
Hinted at this in my first comment but I do also agree it would be nice to use `items` - we seem to special case it a lot (I've done so in groupby) I think just to work around the fact that it can return a DataFrame for duplicate labels

If we didn't do that maybe we could just enumerate over `df.items()` and have one way of doing things

Reply:
> Hinted at this in my first comment

Oh! you meant DataFrame.items!  I thought you were referring to BlockManager.items, which matches DataFrame.columns

Reply:
yes (with an enumerate on top if you need the i)

Reply:
updated to use enumerate and .items

Reply:
This looks good. Only comment I have is that this probably will fail with duplicate column labels if not already tested.

Reply:
DataFrame.items has a check for columns.is_unique; is that check not enough?

Reply:
Ah thanks; misunderstanding on my end